### PR TITLE
Add main governance document formalizing project leadership structure

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
This PR adds the proposed main Governance Document for the Spyder project, along with a supporting list of its relevant People (to be revised as necessary). As described at some length therein, it is directly based on the [Pandas Main Governance Document](https://github.com/pandas-dev/pandas-governance/blob/f7352be07872048a07be3d3614ae142e004568df/governance.md) that was given as an exemplar by NumFOCUS of what was required, which is a fairly good fit for as, as it relatively closely matches our existing governance model. I then modified it as required to eliminate sections and provisions that aren't relevant for Spyder, adding in various bits that are, modifying it as needed to reflect the differences between the two projects, clarifying some segments left ambiguous in the original document, and addressing some editorial issues.

Also, I've added (also modeled closed on Pandas) a corresponding People List, listing the people in various positions relevant to the Governance Document, which will serve as a public record of such as required by NumFOCUS's transparency requirements and also list the required five NumFOCUS contacts, among others. In some places the assignments of people to roles are sketchy or incomplete; we can discuss this further at the meeting and make the necessary adjustments as needed.

Hopefully we can all try to look it over before then, but we can kick off the formal discussion with the team at our weekly developer meeting, and go from there. We'll need to have this voted on, approved and merged before we can start the process to become a NumFOCUS Fiscally Sponsored project, which we can talk about at the meeting as well, but we should also be sure we are all happy with it and any feedback has been incorporated.

### CODE REVIEW

File: governance.md
Comment (jitseniesen) [inline governance.md]:
```suggestion
Potential Core Developers are nominated by a Core Developer and voted upon by the Core Developers after asking if the potential Core Developer is interested and willing to serve in that capacity.
```

**Review by jitseniesen — COMMENTED** [review]
Looks fine at first sight.

File: governance.md
Comment (CAM-Gerlach) [reply governance.md]: Thanks, I fixed that and revised the sentence a bit to make it shorter, add a missing comma and reduce repetition.

**Comment by CAM-Gerlach** [comment]
Hey @jitseniesen , do you want to be added to the NumFOCUS and/or CoC teams in the People list? I would have, but I wasn't sure how active you were. BTW, are you able to make the next developer meeting this Thursday at 20:00 UTC on the Spyder Zoom?

**Comment by jitseniesen** [comment]
> Hey @jitseniesen , do you want to be added to the NumFOCUS and/or CoC teams in the People list? I would have, but I wasn't sure how active you were. BTW, are you able to make the next developer meeting this Thursday at 20:00 UTC on the Spyder Zoom?

I am planning to attend the meeting, we can have a chat about it then.

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
The Project wants to encourage a diverse array of backgrounds, viewpoints and talents on the Core Developer team, which is why code is not defined as the sole metric on which Core Developers membership will be evaluated.
```

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
Retired Core Developers members will be also included on the People list, acknowledging the period during which they were active in the Core Developers Team.
```

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
Organization Members are Contributors who, due to a reasonable and appropriate need (such as having a formal relationship with the Spyder Project, or being considered or mentored for eventual nomination as a Core Developer), are added as Members of the Spyder-IDE Organization.
```

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
They may be granted permissions of "Triage" or lower on the `spyder` or `spyder-kernels` repositories, or "Maintain" or lower on any other repository, and be added to any Team with the same permissions.
```

**Review by mrclary — DISMISSED** [review]

File: governance.md
Comment (dalthviz) [inline governance.md]: Fix link to QtPy:
```suggestion
Examples of Project Software include the main [Spyder](https://github.com/spyder-ide/spyder) scientific Python development environment, the [Spyder-Kernels](https://github.com/spyder-ide/spyder-kernels) support package, the [QtPy](https://github.com/spyder-ide/qtpy) Qt abstraction layer, and the [Spyder-Docs](https://github.com/spyder-ide/spyder-docs) documentation.
```

File: governance.md
Comment (dalthviz) [inline governance.md]: Just to settle that the Core Developers are part of the Community:
```suggestion
In general, all Project decisions are made through consensus among the Core Developers with input from the rest off the Community.
```

File: governance.md
Comment (dalthviz) [inline governance.md]:
```suggestion
When private communications and decisions are needed, the Core Developers will do its best to summarize those to the rest of the Community after eliding personal/private/sensitive information that should not be posted to the public internet.
```

File: governance.md
Comment (dalthviz) [inline governance.md]:
```suggestion
Unless specifically required, all Core Developer and Organization discussions and activities should be public and done in collaboration and discussion with the Project Contributors and Community in general.
```

**Review by dalthviz — COMMENTED** [review]
Hi! Left some suggestions (fixing a link and some phrasing). Otherwise this LGTM :+1:

File: governance.md
Comment (CAM-Gerlach) [reply governance.md]:
```suggestion
In general, all Project decisions are made through consensus among the Core Developers with input from the rest of the Community.
```

Good point—I just fixed a typo in the suggestion ("of", not "off"; you got it right the other spot)

**Review by dalthviz — DISMISSED** [review]
:+1:

**Comment by jitseniesen** [comment]
> Hey @jitseniesen , do you want to be added to the NumFOCUS and/or CoC teams in the People list?

I am happy to be added to the NumFOCUS team.

File: people.md
Comment (CAM-Gerlach) [inline people.md]:
```suggestion
* Jitse Niesen ([@jitseniesen](https://github.com/jitseniesen))
```

**Review by dalthviz — DISMISSED** [review]

**Review by mrclary — DISMISSED** [review]

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
Contributors with a sustained, formal relationship to the project are added as Organization Members, part of the Spyder-IDE Organization.
```

Seems like a typo?

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
* If an Owner or their account are believed to be physically or mentally compromised, either on their request or with the concurrence of the other two Owners, they may be temporarily removed for a period of up to two weeks and then restored, unless two thirds of the remaining active Core Developers vote for their permanent removal.
```

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
The Core Developers will have a private Discord channel that will be used sparingly when a specific matter requires privacy.
```

File: governance.md
Comment (ccordoba12) [inline governance.md]:
```suggestion
Non-conformance with the terms of the governance documents shall be reported to the Core Developers, either through public (issue on the Governance and Guidelines repository) or private (Core Developer Discord) channels as deemed appropriate.
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks a lot @CAM-Gerlach for your work on this and all other people that participated with your reviews and comments in this PR.

This looks almost ready to me, except for the very minor suggestions I left below.

File: CHANGELOG.md
Comment (CAM-Gerlach) [inline CHANGELOG.md]:
```suggestion
## 2022-10-13
```

Set final planned merge date

**Comment by CAM-Gerlach** [comment]
@PierreRaybaut if you'd like to give feedback on this, please feel free as well. I figured I'd ping you since you were Spyder's original creator, and your name is also mentioned a couple times in here. Thanks!

File: governance.md
Comment (mrclary) [inline governance.md]:
```suggestion
Per-repository Collaborators are Contributors who, thanks to their record of contributions to a specific repository, to further deepen collaboration, or a specific need for their expertise, are given a Role on a particular Spyder-IDE Organization repository.
```

File: governance.md
Comment (mrclary) [inline governance.md]:
```suggestion
This section concerns the various Teams, comprised of Core Developers and other Organization Members, for the purposes of governance and for coordination and specialization within the Organization.
```

**Review by mrclary — CHANGES_REQUESTED** [review]
I left a few grammatical suggestions.

**Review by mrclary — DISMISSED** [review]

**Comment by PierreRaybaut** [comment]
> @PierreRaybaut if you'd like to give feedback on this, please feel free as well. I figured I'd ping you since you were Spyder's original creator, and your name is also mentioned a couple times in here. Thanks!

Hey @CAM-Gerlach , thanks for alerting me to the existence of this PR. I will take a closer look to this document and give feedback.

**Comment by CAM-Gerlach** [comment]
Thanks. Just FYI, we kinda need it to be approved and merged by the time our NumFOCUS application is reviewed (submitted on or about Saturday, Oct 15), which will allow us to be officially supported by the NumFOCUS nonprofit org to help fund the project and ensure it continues to thrive (given recent-ish events), so if you could try to get your feedback in before then (ideally, before or at our regular Thursday afternoon meeting, if you'd like to attend), that would be super :slightly_smiling_face: Especially since everyone needs to approve the final version again if any changes are requested, which can be a bit like herding cats, so the sooner the better.

**Comment by PierreRaybaut** [comment]
> > @PierreRaybaut if you'd like to give feedback on this, please feel free as well. I figured I'd ping you since you were Spyder's original creator, and your name is also mentioned a couple times in here. Thanks!
> 
> Hey @CAM-Gerlach , thanks for alerting me to the existence of this PR. I will take a closer look to this document and give feedback.

@CAM-Gerlach : that is done.

This is a fine document, well done.
(I've just added my humble contribution)

**Comment by CAM-Gerlach** [comment]
Thanks @PierreRaybaut !

**Review by mrclary — APPROVED** [review]

**Review by dalthviz — APPROVED** [review]

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now.

**Review by jitseniesen — APPROVED** [review]

**Review by impact27 — APPROVED** [review]
Looks good

**Review by jitseniesen — APPROVED** [review]

**Review by ccordoba12 — APPROVED** [review]

**Review by dalthviz — APPROVED** [review]

**Comment by CAM-Gerlach** [comment]
Looks like we all approve now, and GitHub has picked it up! We can formally hit the <kbd>Merge</kbd> button in our meeting in 30 minutes time.

**Pull request merged**
